commons/ava_testsuite/staking_gecko_network_tests.go

```diff
@@ -191,21 +179,266 @@ func (test FiveNodeStakingNetworkGetValidatorsTest) GetTimeout() time.Duration {
+			if serviceId != bootServiceId {
+				bootNodeId := allNodeIds[bootServiceId]
+				acceptableNodeIds[bootNodeId] = true
+			}
+		}
+
+		if _, found := bootServiceIds[serviceId]; found {
+			// Boot nodes should have the original node, one of the duplicates, and MAY have the duplicate nodes
+			acceptableNodeIds[allNodeIds[NODE_SERVICE_ID]] = true
+			acceptableNodeIds[badServiceNodeId1] = true
+			acceptableNodeIds[badServiceNodeId2] = true
+			verifyExpectedPeers(context, serviceId, allGeckoClients[serviceId], acceptableNodeIds, len(originalServiceIds) - 1, true)
+		} else {
+			// The original non-boot node should have exactly the boot nodes
+			verifyExpectedPeers(context, serviceId, allGeckoClients[serviceId], acceptableNodeIds, len(bootServiceIds), false)
+		}
+	}
+	logrus.Info("Verified that original nodes are still connected to each other")
+
+	// Now, kill the first dupe node to leave only the second (who everyone should connect with)
+	logrus.Info("Removing first node with duplicate ID...")
+	if err := castedNetwork.RemoveService(badServiceId1); err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Could not remove the first service with duped node ID"))
+	}
+	delete(allServiceIds, badServiceId1)
+	delete(allGeckoClients, badServiceId1)
+	delete(allNodeIds, badServiceId1)
+	logrus.Info("Successfully removed first node with duplicate ID, leaving only the second")
+
+	// Now that the first duped node is gone, verify that the original node is still connected to just boot nodes and
+	//  the second duped-ID node is now accepted by the boot nodes
+	logrus.Info("Verifying that the network has connected to the second node with a previously-duplicated node ID...")
+	for serviceId, _ := range allServiceIds {
+		acceptableNodeIds := make(map[string]bool)
+
+		// All nodes should have the boot nodes (though a boot node won't have itself)
+		for bootServiceId, _ := range bootServiceIds {
+			if serviceId != bootServiceId {
+				bootNodeId := allNodeIds[bootServiceId]
+				acceptableNodeIds[bootNodeId] = true
+			}
+		}
+
+		// Boot nodes should have all nodes
+		if _, found := bootServiceIds[serviceId]; found {
+			acceptableNodeIds[allNodeIds[NODE_SERVICE_ID]] = true
+			acceptableNodeIds[badServiceNodeId2] = true
+			verifyExpectedPeers(context, serviceId, allGeckoClients[serviceId], acceptableNodeIds, len(allServiceIds) - 1, false)
+		} else {
+			verifyExpectedPeers(context, serviceId, allGeckoClients[serviceId], acceptableNodeIds, len(bootServiceIds), false)
+		}
+	}
+	logrus.Info("Verified that the network has settled on the second node with previously-duplicated ID")
+}
+
+func (f FiveNodeStakingNetworkDuplicateIdTest) GetNetworkLoader() (testsuite.TestNetworkLoader, error) {
+	return getFiveNodeStakingLoader()
+}
+
+func (f FiveNodeStakingNetworkDuplicateIdTest) GetTimeout() time.Duration {
+	return 120 * time.Second
+}
+
 // =============== Helper functions =============================
 
 // TODO TODO TODO Rename this
```

Comment:
thoughts on a better name for this?

Reply:
so basically, once we merge your PR, then mine, I'm going to rename everything in this file to move away from the `FiveNode...` convention (since it's actually not 5)